### PR TITLE
CIV-5345 Full Admit + Pay Immediately

### PR DESCRIPTION
**Pull request by portizdev**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-5345
https://tools.hmcts.net/jira/browse/CIV-9059

### Change description ###
Adapt Full Admit + Pay Immediately LR vs LR to operate in LR vs LiP (XUI)
Common function to get Payment Date when Claim Admitted

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by laperla11 — APPROVED** [review]

**Review by kamilewandowski — DISMISSED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            .respondToClaimAdmitPartLRspec(
+                RespondToClaimAdmitPartLRspec.builder()
+                    .whenWillThisAmountBePaid(whenWillPay)
+                    .build()
+            )
+            .build();
+        //When
+        LocalDate payDate = paymentDateService.getPaymentDateAdmittedClaim(caseData);
+        //Then
+        org.assertj.core.api.Assertions.assertThat(payDate).isEqualTo(whenWillPay);
```
Comment (PaulH-HMCTS) [inline src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java:40]: Would be more readable to static import assert methods

File: src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            )
+            .build();
+        //When
+        LocalDate result = paymentDateService.getPaymentDateAdmittedClaim(caseData);
+        //Then
+        org.assertj.core.api.Assertions.assertThat(result).isEqualTo(whenWasPaid);
+    }
+
+    @Test
+    void shouldReturRespondent1ResponseDatePlus5Days_whenRespondent1ResponseDateArePresent() {
```
Comment (PaulH-HMCTS) [inline src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java:59]:
```suggestion
    void shouldReturnRespondent1ResponseDatePlus5Days_whenRespondent1ResponseDateArePresent() {
```

File: src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            .respondent1ResponseDate(LocalDateTime.now())
+            .build();
+        //When
+        LocalDate result = paymentDateService.getPaymentDateAdmittedClaim(caseData);
+        //Then
+        org.assertj.core.api.Assertions.assertThat(result).isEqualTo(whenWillPay);
+    }
+
+    @Test
+    void shouldReturNull_whenAnythingIsSettled() {
```
Comment (PaulH-HMCTS) [inline src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java:73]:
```suggestion
    void shouldReturnNull_whenAnythingIsSettled() {
```

**Review by PaulH-HMCTS — DISMISSED** [review]

**Review by laperla11 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java (hunk 3 above)
Comment (portizdev) [reply src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java]: Done

File: src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java (hunk 1 above)
Comment (portizdev) [reply src/test/java/uk/gov/hmcts/reform/civil/service/PaymentDateServiceTest.java]: Done. Thank you for the comments

**Review by neeta-hmcts — APPROVED** [review]

**Review by rajakm — APPROVED** [review]

**Review by sankhajuria — APPROVED** [review]

**Pull request merged**